docs/planning/otel/OTelWebSdk.md

```diff
@@ -0,0 +1,1567 @@
+### Shared Resource Implementation
+
+```typescript
+class SharedExportQueue implements IExportQueue {
+  private _queue: ITelemetryItem[] = [];
+  private _maxSize: number;
+  private _batchTimer: ITimerHandle | null = null;
+  private _processing: boolean = false;
+
+  constructor(maxSize: number = 1000) {
+    this._maxSize = maxSize;
+  }
+
+  enqueue(item: ITelemetryItem): void {
+    if (this._queue.length >= this._maxSize) {
+      // Drop oldest items or apply sampling
+      this._queue.shift();
+    }
+    
+    this._queue.push(item);
+    this._scheduleBatch();
+  }
+
+  async flush(): Promise<void> {
+    if (this._processing || this._queue.length === 0) {
+      return;
+    }
+
+    this._processing = true;
+    const batch = this._queue.splice(0, 100); // Process in batches
+
+    try {
+      await this._processBatch(batch);
+    } finally {
+      this._processing = false;
+      
+      // Continue processing if more items arrived
+      if (this._queue.length > 0) {
+        this._scheduleBatch();
+      }
+    }
+  }
+
+  private _scheduleBatch(): void {
+    if (this._batchTimer) return;
+    
+    this._batchTimer = setTimeout(() => {
+      this._batchTimer = null;
+      this.flush();
+    }, 5000); // 5-second batching interval
+  }
+
+  private async _processBatch(items: ITelemetryItem[]): Promise<void> {
+    // Group items by endpoint and send efficiently
+    const groups = this._groupByEndpoint(items);
+    
+    const exportPromises = Array.from(groups.entries()).map(([endpoint, groupItems]) =>
+      this._exportToEndpoint(endpoint, groupItems)
+    );
+    
+    await Promise.all(exportPromises);
+  }
+}
+
+class SharedConnectionPool implements IConnectionPool {
+  private _connections: Map<string, IConnection> = new Map();
+  private _maxConnections: number;
+
+  constructor(maxConnections: number = 2) {
+    this._maxConnections = maxConnections;
+  }
+
+  getConnection(endpoint: string): IConnection {
+    if (!this._connections.has(endpoint)) {
+      if (this._connections.size >= this._maxConnections) {
+        // Reuse least recently used connection
+        const lruEndpoint = this._findLRUConnection();
+        this._connections.delete(lruEndpoint);
+      }
+      
+      this._connections.set(endpoint, this._createConnection(endpoint));
+    }
+    
+    return this._connections.get(endpoint)!;
+  }
+
+  private _createConnection(endpoint: string): IConnection {
+    return new OptimizedConnection(endpoint, {
+      keepAlive: true,
+      maxConcurrency: 6,
+      retryConfig: {
+        maxRetries: 3,
+        backoffMs: 1000
+      }
+    });
+  }
+}
+```
+
+## Factory Setup and Distribution Strategies
+
+The OTelWebSdk factory requires different strategies for sharing and distribution depending on how teams consume the SDK. This section details how the factory handles NPM vs CDN loading scenarios and prevents conflicts.
+
+### **NPM Distribution: Isolated Factories with Optional Global Registry**
```

Comment:
Another section that I think just needs to go, there is too much wrong here to try and "comment" on.

Reply:
The only relevant bit is that we need to have a strategy to "handle" this, but this is mostly covered by using IoC and factory functions to keep version isolation, not all of this "use globals" to track versions. The only think we "might" (thought only) want is a simple version list stating how that version was loaded (but we can partially derive this by looking for the loaded &lt;script /&gt; tags